Added free-game-checker to plugins

danivdwerf opened the pull request:
# Free Game Checker
This plugin hourly check a list of predefined services if any free games are available. If a new game is available and the user has enabled toasts, it will show a toast. Within the plugin is a list of available games per Service, which will navigate you to the store page when clicked. The user is also able to toggle toasts per service. As for now, only the Epic Games Service is available.

## How is this different from [WerWolv/SDH-FreeGames](https://github.com/WerWolv/SDH-FreeGames)?
- This plugin doesn't call external API's every second, but rather every hour.
- This plugin only sends a notification once per game, rather than every day.
- This plugin allows users to opt-out of notifications.
- This plugin navigates user to the store page when a game is clicked.
- This plugin is set up to easily integrate services other than the Epic Game Store

## Checklist:

### Developer Checklist

- [x] I am the original author or an authorized maintainer of this plugin.
- [x] I have abided by the licenses of the libraries I am utilizing, including attaching license notices where appropriate.

### Plugin Checklist

- [x] I have verified that my plugin works properly on the Stable and Beta update channels of SteamOS.
- [x] I have verified my plugin is unique or alternatively provides more/alternative functionality to a similar plugin already on the store.
### Plugin Backend Checklist

- **No**: I am using a custom backend other than Python.
- **No**: I am using a tool or software from a 3rd party FOSS project that does not have it's dependencies [statically linked](https://en.wikipedia.org/wiki/Static_library).
- **No**: I am using a custom binary that has all of it's dependencies statically linked.

<!-- The following section is for testers and maintainers, please do not modify it unless you don't meet the criteria for testing on the preview channel. -->

## Testing

- [ ] Tested on SteamOS Stable Update Channel.
- [x] Tested on SteamOS Beta Update Channel.

TopHatHipster commented:
If you could provide a .zip release to directly install onto the Steam Deck, I'll be happy to test it ASAP. Will have to still set up a local dev environment to compile next time, so I don't have to request a pre-compiled release.

TrainDoctor commented:
As TopHat hit on above,
Given recent changes by valve and a mistake on our part for pnpm versions, please update pnpm to >= v8 and bump your the DFL version for you plugin to minimum 3.20.4 or the latest release so that your plugin can recieve testing on Beta.

danivdwerf commented:
@TopHatHipster Would a link to the release page with the app assets be sufficient? I've updated DFL to 3.20.5 for this release:
https://github.com/danivdwerf/free-game-checker/releases/tag/0.0.1

TopHatHipster commented:
@danivdwerf Nope, I refer to a compiled/"store ready" version. Downloading the source code as a .zip and trying to install it through the URL method on Decky Loader does not install the plugin, as it's not packaged correctly.

You can build a release zip by executing the `pnpm i` and `pnpm run build` commands on the project's directory, and zipping up the output of that without any other folders inside of it (as in: selecting the files and zip it up, instead of zipping up a single folder that contains the entire output of the build command).

danivdwerf commented:
@TopHatHipster The release does not contain any source files, just the files you've mentioned. I'll add the same zip to this comment again:
[free-game-checker.zip](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/11182884/free-game-checker.zip)

TopHatHipster commented:
@danivdwerf After installing the release of that zip through Decky Loader (the one on the releases page is just the source code, it doesn't have any other types of zips included as shown here in the attached picture) and rebooting the Deck, it seems to work perfectly fine:

- Loads in the freebies of the Epic Games Store just fine (although it incorrectly states Borderlands 3's season pass is currently free, which doesn't seem to be the case?)
- Notifications of each are given just fine when having notification enabled.
- Notifications will not appear when notifications are disabled

Tested it on:
- Steam Deck Beta Client (1680827821/Apr. 7 update)
- SteamOS Beta Candidate (3.4.6/build 20230302.1)
- Decky Loader v2.6.5-pre2

Seems to be ready to get shipped to the Testing Store, once the Borderlands 3 Season Pass issue is fixed. 👍
Although it may be the same as the current Free Games plugin, I'd personally argue your differences make it a better plugin overall to be present on the plugin store (as long as it's maintained enough to remain functional).

AAGaming00 commented:
I feel like this would be better as a PR/merge to Wolv's plugin rather than a separate plugin

AAGaming00 commented:
This "calling apis every second" is probably a bug with your plugin @WerWolv

TopHatHipster commented:
> I feel like this would be better as a PR/merge to Wolv's plugin rather than a separate plugin

Not to mention, Wolv's plugin needs a DFL update as it broke on the beta Steam Client.

danivdwerf commented:
> Seems to be ready to get shipped to the Testing Store, once the Borderlands 3 Season Pass issue is fixed. 👍

Oh good spot! I thought I filtered out all that crap, but I'll get on it.

> (as long as it's maintained enough to remain functional)

I'll definitely take the time to periodically update the underlying modules, since it benefits me too if the plugin is working. I've already written the logic for also supporting the GOG store, but I didn't want to over complicate things for you guys for the initial release so I'll wait with that PR since this version has been tested now.

@AAGaming00
> This "calling apis every second" is probably a bug with your plugin @WerWolv

I don't know, but [it seems like its by design](https://github.com/WerWolv/SDH-FreeGames/blob/main/src/index.tsx#L116)

> I feel like this would be better as a PR/merge to Wolv's plugin rather than a separate plugin

It does replace 100% of the code, but if they're up for it sure. Though if I'm looking at their repo it would seem that I'm able to update and add more functionality more frequently

AAGaming00 commented:
reading that code further, it doesn't call the API unless the current day has changed

WerWolv commented:
@danivdwerf I can give you access to my repo if you'd like. I'm really not attached to any of my shitty code lol
Also it's not calling any APIs every second, it's checking if a day has passed every second. That could definitely be lowered to once every minute or so though

danivdwerf commented:
> I can give you access to my repo if you'd like

@AAGaming00 @TopHatHipster  Would that work for you guys?

danivdwerf commented:
@WerWolv I would love to have access to the repository to create a new release. I've fixed the non free games slipping through the filters on the Epic Games Store, added gog.com and added some settings for notifications.

TrainDoctor commented:
__Note: This plugin is a unique scenario where a member of the team needs to work with the developer on this PR. It will be granted a bit more time and leeway but will ultimately have to be treated the same.__

In case you are not a part of the SDH Discord, I am posting this in the PRs for all currently grandfathered in plugins as there has been an update to the review and submissions process to the plugin database. You can view updated version [here](https://wiki.deckbrew.xyz/en/plugin-dev/review-and-testing).

The simplest way to explain the change is this: Plugin/Plugin Update submissions will now have a max of about a month to receive testing on:

A) Stable or Beta if they do not meet the criteria for using a custom backend
or
B) Preview if they do meet the criteria for using a custom backend.

Please make sure to check whether or not your plugin meets the definition of having a custom backend by reading the updated review and testing page.

We also are able to provide help finding testers for users who join the SDH [discord](deckbrew.xyz/discord).
Here are the more detailed specifics of how we will be handling this going forward.
Specifically, if we do not see any movement towards making sure the plugin/plugin update is being tested and a testing review is/will be submitted we will:

1. Contact the plugin developer and offer to help find testers for them.
  - If we receive no response or a response that the team understands as moving forward with testing, we reserve the right to close the PR and pull the plugin/plugin update after 2 weeks.
2. Contact the plugin developer again and more directly determine a firm timeline on testing.
  - This should only occur around 3 weeks after submission, at this point if we have not exercised our right to close the PR after 2 weeks, you can expect the PR will be closed and the plugin/plugin update will be pulled from the testing store.

We are open to feedback and understand that life and work  happen. Any PRs closed due to these occurrences will be allowed to re-open as soon as the developer is able to return to them.

TrainDoctor commented:
In light of most recent comments, closing this PR.

danivdwerf commented:
@WerWolv I finally took the time to upgrade everything to fix the modpanel errors and have the plugin working for the latest version fo SteamDeck. Could you give me access to the repository so I can create a new PR?
